addons/udes_stock/tests/test_scanned_by.py

```diff
@@ -0,0 +1,64 @@
+from .common import BaseUDES
+from odoo.api import Environment
+
+
+class TestScannedBy(BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        """Setup a picking to test user scanned by is correct"""
+        super().setUpClass()
+        # Setup user
+        cls.test_company = cls.create_company("test_company")
+        cls.test_user = cls.create_user(
+            "test_user",
+            "test_user_login",
+            company_id=cls.test_company.id,
+            company_ids=[(6, 0, cls.test_company.ids)],
+        )
+
+        cls.other_user = cls.create_user(
+            "other_user",
+            "other_user_login",
+            company_id=cls.test_company.id,
+            company_ids=[(6, 0, cls.test_company.ids)],
+        )
+
+        test_user_env = Environment(cls.env.cr, uid=cls.test_user.id, context=cls.env.context)
+
+        # Setup picking
+        cls.create_quant(cls.apple.id, cls.test_stock_location_01.id, 10)
+        cls.create_quant(cls.banana.id, cls.test_stock_location_02.id, 20)
+        picking = cls.create_picking(cls.picking_type_internal)
+        move = cls.create_single_move(cls.apple, 10, picking)
+        move = cls.create_single_move(cls.banana, 20, picking)
+        picking.action_confirm()
+        picking.action_assign()
+
+        cls.picking = picking.with_env(test_user_env)
+
+    def test_records_real_id_on_move_line_update(self):
```

Comment:
I'd name the method `test_records_original_user_id_on_move_line_update`, "real" doesn't seem to mean anything here.
In the docstring 
- u_done_by -> u_done_by_id?
- required -> used?
- and be reported on -> can be reported on?

Reply:
Apologies, an even better name would be

`test_records_original_user_id_on_move_line_update_with_sudo`

Reply:
fixed